cmd/create/machinepool/nodepool.go

```diff
@@ -222,3 +239,44 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		r.Reporter.Infof("To view all machine pools, run 'rosa list machinepools -c %s'", clusterKey)
 	}
 }
+
+func getSubnetFromAvailabilityZone(cmd *cobra.Command, r *rosa.Runtime, isAvailabilityZoneSet bool,
+	cluster *cmv1.Cluster) (string, error) {
+
+	privateSubnets, err := r.AWSClient.GetVPCPrivateSubnets(cluster.AWS().SubnetIDs()[0])
+	if err != nil {
+		return "", err
+	}
+
+	// Fetching the availability zones from the VPC private subnets
+	availabilityZones := []string{}
+	subnetsMap := make(map[string]string)
+	for _, privateSubnet := range privateSubnets {
+		if privateSubnet.AvailabilityZone != nil && privateSubnet.SubnetId != nil {
+			availabilityZones = append(availabilityZones, *privateSubnet.AvailabilityZone)
+			subnetsMap[*privateSubnet.AvailabilityZone] = *privateSubnet.SubnetId
+		}
+	}
+	availabilityZone := cluster.Nodes().AvailabilityZones()[0]
+	if !isAvailabilityZoneSet && interactive.Enabled() {
+		availabilityZone, err = interactive.GetOption(interactive.Input{
+			Question: "AWS availability zone",
+			Help:     cmd.Flags().Lookup("availability-zone").Usage,
+			Options:  availabilityZones,
```

Comment:
Just to simplify it for us and the users I would suggest following the same pattern as classic ROSA.
1. If the user chooses AZ, we go for the cluster's subnet belonging to the AZ, which was provided on day 1.
2. Otherwise, the user can select a subnet that belongs to the VPC, choosing AZ implicitly.

I think it covers all cases, please let me know if I missed anything.

Reply:
But I guess we will hit again the scenario where users in day 2 add a new subnet to that AZ, then perhaps @pvasant makes sense to detect that there is more than one and fall into the subnet selection.

Reply:
I agree.

What I meant is that if you fall into a subnet selection, it was possible to select a subnet in the first place.
For example:
1. ? Select subnet for a single AZ machine pool (optional): No
2. ? AWS availability zone:  [Use arrows to move, type to filter, ? for more help]
  us-west-2c
  us-west-2b
> us-west-2a
3. ? Subnet ID:  [Use arrows to move, type to filter, ? for more help]
> subnet-06023544bf8e5b199 (us-west-2c)

Reply:
Ah, I think that if the subnet is selected in the first place, it shouldn't ask you to select AZ.
But yes, if you select AZ, and it detects more than one Subnet, it will ask again for a Subnet.

Reply:
I discussed this with @pvasant and she said it's ok to limit the AZ selection to the cluster's AZs.
So if the user wants to use another AZ they have to provision a subnet there and select it.

It's more of UX experience than functionality.
I would recommend sticking with the way it's done in classic ROSA, but it's your call.
Let me know if I missed anything.

Reply:
> So if the user wants to use another AZ they have to provision a subnet there and select it.

So, what's the usual flow in rosa classic? users edit the cluster to add an additional subnet there?

Reply:
But, do users need to add any new subnet in the cluster even if that subnet exists in the same AZ?

Perhaps these two flows are not 100% clear to me.

Reply:
Anyway, I'm ok with the suggestions @oriAdler @pvasant I have a pending backend work for Day 1 that will enable the pending updates in the cluster to check this.

Reply:
> limit the AZ selection to the cluster's AZs

There _are_ no cluster AZs, so this doesn't really make sense ¯\_(ツ)_/¯ 

IMO, we should take whatever input is provided by the customer, and then validate, e.g.,

- `rosa create machinepool --subnet subnet-1234` -> `subnet-1234` exists and is private -> OK
- `rosa create machinepool --subnet subnet-bad` -> `subnet-bad` doesn't exist -> ERROR
- `rosa create machinepool --availability-zone us-east-1a` -> `us-east-1a` has more than 1 private subnet -> interactive mode
- `rosa create machinepool --availability-zone us-east-1b` -> `us-east-1b` has 1 private subnet -> OK
- `rosa create machinepool --availability-zone us-east-1c` -> `us-east-1c` has 0 private subnets -> ERROR

There shouldn't need to be any additional consideration or validation on subnets/AZs that were provided on day-1.

This is a restriction on _classic_ ROSA only because of the Control Plane node placement, but that restriction doesn't exist in HCP

Reply:
My only concern right now is we should then allow giving subnet and az together in the command else automation will not work for these scenarios. 

rosa create machinepool --availability-zone us-east-1a -> us-east-1a has more than 1 private subnet -> interactive mode

Reply:
@lucasponce i think we need to support giving giving both subnet and AZ for HS

Reply:
@pvasant +1 I can update it that way.

Reply:
Or perhaps, what about

rosa create machinepool --availablility-zone us-east-1a -> Move to interactive only if --interactive is set, otherwise, to fail ?

Reply:
I m not sure if i understand . requirement is there should  be a way to run without interactive mode. they pass subnet then az is not mandatory since we can derive if they pass az subnet is mandatory to run in non interactive mode right

Reply:
Ok, perhaps I was overthinking these two cases:

- When subnet and AZ are both passed, if subnet doesn't belong to AZ, should it fail? or should ignore the AZ parameter?
- When AZ is passed, if there are more than one subnet and it's not in interactive mode should fail, right?

Reply:
IMO, I think the two (AZ and subnet) should be mutually exclusive, and provide a helpful error message when both provided.

If you want to use a specific _AZ_, then you should provide that.

If you want to use a specific _subnet_, then you should provide that.

Providing both, seems redundant and leads to confusion as illustrated ¯\_(ツ)_/¯

Reply:
Thanks, @wgordon17 @pvasant @oriAdler; I think we are very close to addressing this one.

So, then if the user specifies an AZ, and this AZ has more than one subnet:
- If it's in interactive mode, it should ask to select a subnet.
- If not, it should fail.

Am I right?

Reply:
Not sure about the failure, should we fall into interactive mode in this case?

Reply:
The current tests & scenarios that these PR implements are described https://github.com/openshift/rosa/pull/1048#issue-1564836204

I also suggest that perhaps we can progress with those (if there are no blockers), and we can refine them in iterative steps if that looks oks